app/javascript/packs/MainApp/components/RefereeProfile.jsx

```diff
@@ -148,12 +155,60 @@ class RefereeProfile extends Component {
       .catch(this.setErrorStateFromBackendData);
   };
 
-  hasPassedTest(level) {
-    const { referee: { certifications } } = this.state;
+  hasPassedTest = (level) => {
+    const { referee: { certifications } } = this.state
 
     return certifications.some(({ level: certificationLevel }) => certificationLevel === level)
   }
 
+  handlePaymentSuccess = (payment) => {
+    const { paid } = payment
+
+    if (paid) {
+      this.setState({ paymentSuccess: true })
```

Comment:
You should clear the other states. If you first cancel and then pay successfully you currently get both messages.